src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -346,15 +347,15 @@ static ride_list_item* window_new_ride_iterate_over_ride_group(
 {
     bool buttonForRideTypeCreated = false;
     bool allowDrawingOverLastButton = false;
-    uint8_t* rideEntryIndexPtr = get_ride_entry_indices_for_ride_type(rideType);
 
     char preferredVehicleName[DAT_NAME_LENGTH + 1];
     safe_strcpy(preferredVehicleName, "        ", sizeof(preferredVehicleName));
 
     // For each ride entry for this ride type
-    while (*rideEntryIndexPtr != RIDE_ENTRY_INDEX_NULL)
+    auto& objManager = OpenRCT2::GetContext()->GetObjectManager();
+    auto rideEntries = objManager.GetAllRideEntries(rideType);
```

Comment:
`const auto&` as the method returns a reference. Applies to all other calls

Reply:
Thanks, very easy to make this mistake.